### PR TITLE
fix(item-divider): update design to match native iOS

### PR DESCRIPTION
**Pull request by brandyscarney**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

Updates to match native iOS design

Screen Shot 2020-03-23 at 6 46 19 PM

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by brandyscarney** [comment]
@liamdebeasi Can you review my thought process behind removing this CSS from both `ios` and `md`: https://github.com/ionic-team/ionic/pull/20854/commits/cef5583cb9bb7aef6b7bd7679607e78d54775faa#diff-9b0cbd35782bfe211ff32e52f0e44974L8

1) The CSS currently doesn't work, because items have inset lines by default and it isn't modifying `--inner-border-width`. So this won't cause a regression by being removed. 
2) If I were to fix this CSS so it _did_ work to match the spec, it would look like this for iOS:

      ```scss
    .item-group-ios ion-item:last-of-type,	
    .item-group-ios ion-item-sliding:last-of-type .item {	
        --inner-border-width: 0;	
        --border-width: 0;	
    }
    ``` 
  
    and this for MD:
  
    ```scss
    .item-group-md ion-item:last-of-type,	
    .item-group-md ion-item-sliding:last-of-type .item {	
      --inner-border-width: 0;	
      --border-width: 0 0 1px 0;	
    }
    ``` 

 3) Adding the above CSS makes it so that the user cannot control the lines by writing `lines="none"` or `lines="full"` on the item if they wanted something other than the default. This results in them having to know that they need to style by `--border-width` and `--inner-border-width` and use a more specific selector than the one above.

So basically - this would not match the spec OOB, but it already doesn't match the spec, and if we want to match the spec OOB it would make it harder for users to override. In the past we've been trying to make sure users have more control over the lines. Ping me if you have questions. Thanks!
